### DISCUSSION TITLE
When defining a new file in a new package, should we optionally check no-littering?

### DISCUSSION THREAD

**Discussion by meedstrom** [emacscollective/no-littering#247] [Support] [answered]
When making a package, that has a new file to write, should I take into account no-littering or just do a PR to this project?

It'd go like this:

```elisp
(defcustom org-node-fakeroam-persist-dir
  (if (bound-and-true-p no-littering-var-directory)
      (file-name-concat no-littering-var-directory "org")
    user-emacs-directory)
  "Directory to hold cached data between sessions."
:type 'directory)
```

EDIT: I just saw that this does quite not follow conventions, my PR will rectify that :)

**Comment by meedstrom** [comment]
I guess it'd make more sense to do a PR, because then it's all visible here, easier to potentially restructure the directory hierarchy from this project's end.

Also safer, since I fear that a conditional clause like above could wind up hardcoding the wrong value after compilation, depending on how the compilation is carried out?

**Reply by tarsius** [reply, marked as answer]
I too think that packages should not try to respect no-littering on their own, for the reasons you give and potentially more.
